packages/firestore/exp/src/api/reference.ts

```diff
@@ -659,6 +667,10 @@ export function onSnapshot<T>(
   reference: Query<T> | DocumentReference<T>,
   ...args: unknown[]
 ): Unsubscribe {
+  if (reference instanceof Compat) {
```

Comment:
Any reason you don't use `cast` here and in other places where you directly use `instanceof`?

Reply:
I am unable to use `cast` in a couple of places where the input type is a union of two distinct class types.